client.go

```diff
@@ -474,19 +474,20 @@ func checkContainerExists(containerID string) bool {
 
 // commitAndExecute commits the given container to a new image, executes the
 // given function and removes the image afterwards.
-func commitAndExecute(f commandFunc, ctx *cli.Context, containerID string) error {
+func commitAndExecute(f commandFunc, ctx *cli.Context, containerID string) (string, error) {
 	client := getDockerClient()
 	// commit container to a new image
 	image, err := client.ContainerCommit(context.Background(), containerID, types.ContainerCommitOptions{})
-	if err != nil {
-		return err
-	}
-	// given commandFunc is executed.
-	f(image.ID, ctx)
 	// remove the image when finished
-	_, err = client.ImageRemove(context.Background(), image.ID, types.ImageRemoveOptions{
+	defer client.ImageRemove(context.Background(), image.ID, types.ImageRemoveOptions{
 		Force:         true,
 		PruneChildren: true,
 	})
-	return err
+	if err != nil {
+		logAndFatalf("\n%s", err)
```

Comment:
Because that would cause checkError to be called with an empty image ID and make the error message provided by humanizeCommandError not nice to read. 
I could of course add a check for an empty string in checkError() and than print a different error message if you prefer that.

Reply:
I have no strong feelings about `checkError()`, so I leave the solution up to you. The problem here is that we're masking some errors but not all. If somebody is calling this function, there will be the expectation that _all_ errors will be returned, not only a subset while the other gets logged.

Reply:
In this case I will just add a check for an empty string to checkError() and print the error message from containerCommit there.